File: Server/src/models/room.ts
import { Message }  from "./message";

export interface Room {
    id: string
    messages: Message[]
    avatars: Map<string, string> // (key: users[i].username, value: users[i].avatar) 
}

export interface CreateRoom {
    id: string
    isPrivate: boolean
}

export interface Invitation {
    id: string,
    username: string
}

File: fat_client/WPFUI/Models/Room.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WPFUI.Models
{
    public class Room
    {
        string id;
        Message[] messages;
        Dictionary<string, string> avatars;

        public Room(string id,
        Message[] messages,
        Dictionary<string, string> avatars)
        {
            this.id = id;
            this.messages = messages;
            this.avatars = avatars;
        }

        public string roomName
        {
            get { return id; }
            set { id = value; }
        }
    }
}
